### PR TITLE
Interface reworking

### PR DESCRIPTION
**Pull request by Ngoguey42**
Improve interface, rename many method/property/parameter to make the library easier to understand.

### Bug fixes
- Fix small bug in `Footprint.__init__` 

### Parameters update
- Add `DataSource.__init__@assert_no_change_on_activation`
- Rename `DataSource.__init__@sr_implicit` to `sr_fallback` (with backward compatibility)
- Rename `DataSource.__init__@sr_origin` to `sr_forced` (with backward compatibility)

### Methods update
- Add `DataSource.copy`
- Rename `DataSource.open_araster` to `aopen_raster ` (with backward compatibility)
- Rename `DataSource.create_araster` to `acreate_raster ` (with backward compatibility)
- Rename `DataSource.open_avector` to `aopen_vector ` (with backward compatibility)
- Rename `DataSource.create_avector` to `acreate_vector ` (with backward compatibility)
- Rename `DataSource.create_recipe_araster` to `acreate_recipe_raster ` (with backward compatibility)

### Properties update
- Rename `Proxy.wkt_origin` to `Proxy.wkt_virtual` (with backward compatibility)
- Rename `Proxy.proj4_origin` to `Proxy.proj4_virtual` (with backward compatibility)
- Add `Proxy.wkt_stored`
- Add `Proxy.proj4_stored`
- Rename `Raster.fp_origin` to `Raster.fp_stored` (with backward compatibility)
- Rename `Vector.extent_origin` to `Vector.extent_stored` (with backward compatibility)
- Add `Vector.bounds_stored`

### Other changes
- Improve `DataSource` documentation
- Rename `RasterPhysical` to `RasterStored`
- Add `buzzard._tools.deprecation_pool` and`buzzard._tools._DeprecationPool`
- Add `buzzard._tools.Singleton` and use it in 3 places
- Partially update the tests to make sure that the `deprecation pool` is unit tested
- Improve error message from `set_data` and `get_data` when illegal interpolation is performed

### CODE REVIEW

File: buzzard/_tools/parameters.py [hunk 1]
```diff
@@ -100,3 +104,100 @@ def _index_generator(clean_indices_generator):
...
+
+        setattr(class_obj, old_name, property(_f))
+
+    def streamline_with_kwargs(self, new_name, old_names, context,
+                               new_name_value, new_name_is_provided, user_kwargs):
+        """Look for errors with a particular parameter in an invocation
+
+        Exemple
+        -------
+        >>> def fn(newname='eee', **kwargs):
```
Comment (azarz) [inline buzzard/_tools/parameters.py]: As said before, I don't really like the 'eee' name for the default value of newname parameter.

File: buzzard/_vector.py [hunk 2]
```diff
@@ -359,8 +369,8 @@ def bounds(self):
 
     @property
     @_tools.ensure_activated
-    def extent_origin(self):
```
Comment (azarz) [inline buzzard/_vector.py]: Why isn't there an extent_stored property?

Comment (Ngoguey42) [reply buzzard/_vector.py]: 10 lines above

File: buzzard/_tools/parameters.py (hunk 1 above)
Comment (Ngoguey42) [reply buzzard/_tools/parameters.py]: You are right, changed

File: buzzard/_vector.py (hunk 2 above)
Comment (azarz) [reply buzzard/_vector.py]: oops! sorry

File: buzzard/_env.py [hunk 3]
```diff
@@ -178,7 +178,7 @@ class Env(object):
     >>> import buzzard as buzz
     >>> with buzz.Env(default_index_dtype='uint64'):
             ds = buzz.DataSource()
-            dsm = ds.open_araster('dsm', 'path/to/dsm.tif')
+            dsm = ds.aopen_raster('dsm', 'path/to/dsm.tif')
```
Comment (Toz3wm) [inline buzzard/_env.py]: still not convinced by this rebranding, should it be for pronunciation reasons only

File: buzzard/_footprint.py [hunk 4]
```diff
@@ -146,8 +146,10 @@ def __init__(self, **kwargs):
         if kwargs:
             raise ValueError('Unknown parameters [{}]'.format(kwargs.keys()))
 
-        if a + b == 0 or d + e == 0:
-            raise ValueError('Scale should not be 0')
+        if a * e - d * b == 0:
+            raise ValueError('Determinent should not be 0: {}'.format(
```
Comment (Toz3wm) [inline buzzard/_footprint.py]: Determinant

**Pull request merged**